modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -677,22 +684,24 @@ export default {
     },
     async onRevertPublishedToPrevious(data) {
       try {
-        const doc = await apos.http.post(`${data.action}/${data._id}/revert-published-to-previous`, {
+        const response = await apos.http.post(`${data.action}/${data._id}/revert-published-to-previous`, {
           body: {},
           busy: true
         });
-        apos.notify('Restored previously published version.', {
+        apos.notify(response.message, {
           type: 'success',
           dismiss: true
         });
         // This handler covers all "undo publish" buttons, so make sure it's
         // for the context document before altering any admin bar state
         // because of it
-        if (doc.aposDocId === (this.context && this.context.aposDocId)) {
+        const aposDocId = data._id.replace(/:.*$/, '');
+        if (data._id.replace(/:.*$/, '') === (this.moduleOptions.context && this.moduleOptions.context._id.replace(/:.*$/, ''))) {
           this.draftIsModified = true;
-          this.publishedAt = null;
-          this.moduleOptions.context.lastPublishedAt = doc.lastPublishedAt;
-          this.refreshOrReload(doc._url);
+          this.lastPublishedAt = response.lastPublishedAt;
```

Comment:
All these things like `lastPublishedAt` and `draftIsModified` should really be on a `this.context` object instead. The admin bar is never published or modified.

Reply:
OK, I'm going to pull out `context` from moduleOptions and shallow-clone it as a deep-watched `context` and we can see how that goes. Deep watches have performance concerns and also can be a maintenance burden because people tend to forget you still can't add a new property and expect reactivity unless using Vue.set. But I agree it will be more readable, let's try it that way, will try this and push.

Reply:
(Fundamentally, the reason we need a reactive `this.context` is because while in principle, we're updating the page and we could refresh the page to get a new option with the latest lastPublishedAt, etc. from the server, nobody wants to refresh the page! So we can view all of these scenarios where context changes in the admin bar as an optimization, but a necessary one for a good user experience. I think the cleanest solution is to take all of `context` into reactive land at startup and stay there, yes.)